### PR TITLE
Create inbound airtime request API endpoint

### PR DESCRIPTION
**Pull request by Lindani**
create models in models.py

### CODE REVIEW

**Comment by Lindani** [comment]
@gsvr Please review before setting up migrations

File: recharges/models.py [hunk 1]
```diff
@@ -4,14 +4,16 @@
 from django.db import models
 
 
-class DummyModel(models.Model):
+class RechargeModel(models.Model):
```
Comment (gsvr) [inline recharges/models.py]: Please call this `Recharge` rather then `RechargeModel`

File: recharges/admin.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 from django.contrib import admin
 
-from .models import DummyModel
+from recharges.models import RechargeModel
 
-admin.site.register(DummyModel)
\ No newline at end of file
+admin.site.register(RechargeModel)
```
Comment (gsvr) [inline recharges/admin.py]: Notice that red thingy at the end of the line? It's complaining that there isn't a blank line at the end of the file.  The easiest way to make this go away is to add the following line to your Sublime Text User settings:
`"ensure_newline_at_eof_on_save": true,`

File: recharges/models.py [hunk 3]
```diff
@@ -4,14 +4,16 @@
...
 
     """
     - Nice docstring please
```
Comment (gsvr) [inline recharges/models.py]: You should add a description of what the Model is for here. Something like:
`Stores all recharge request data.`  We'll probably update this with more info as we go.

**Comment by Lindani** [comment]
@gsvr Please re-review

File: recharges/models.py [hunk 4]
```diff
@@ -4,14 +4,16 @@
...
     """
     id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    product_code = models.CharField(max_length=20)
-    data = HStoreField(null=True, blank=True)
+    amount = models.DecimalField(max_digits=5, decima_places=2)
```
Comment (gsvr) [inline recharges/models.py]: decimal_places typo.
Does max_digits include the decimal_places? So the max recharge would be 999.99?  Or 99999.99?

File: recharges/models.py [hunk 5]
```diff
@@ -4,14 +4,16 @@
...
+    """data = HStoreField(null=True, blank=True)"""
```
Comment (gsvr) [inline recharges/models.py]: It looks like you're trying to comment out this field. The correct way to do that is using the `#` character.  The """ is sometimes used for multiline comments, but mostly for docstrings.  However let's discuss whether we should have this field in the first place tomorrow with Mike.

File: recharges/models.py [hunk 6]
```diff
@@ -4,14 +4,16 @@
...
+    msisdn = models.CharField(max_length=200)
+    author = models.CharField(max_length=200)
```
Comment (gsvr) [inline recharges/models.py]: Let's discuss this field with Mike tomorrow morning.

File: recharges/tests.py [hunk 7]
```diff
@@ -7,7 +7,7 @@
 from rest_framework.authtoken.models import Token
 
 
-from .models import DummyModel
+from recharges.models import RechargeModel
```
Comment (gsvr) [inline recharges/tests.py]: You'll have to remove the `Model` part everywhere.

File: recharges/views.py [hunk 8]
```diff
@@ -1,9 +1,9 @@
 from django.contrib.auth.models import User, Group
-from .models import DummyModel
+from .models import RechargeModel
 from rest_framework import viewsets
 from rest_framework.permissions import IsAuthenticated
-from .serializers import (UserSerializer, GroupSerializer,
-                          DummyModelSerializer)
+from recharges.serializers import (UserSerializer, GroupSerializer,
+                          RechargeModelSerializer)
```
Comment (gsvr) [inline recharges/views.py]: Let's have a chat about pep8 tomorrow, I don't think we've talked about it.

**Comment by Lindani** [comment]
@gsvr Please re-review

File: recharges/models.py [hunk 9]
```diff
@@ -1,19 +1,18 @@
...
-class DummyModel(models.Model):
+class Recharge(models.Model):
 
     """
     - Nice docstring please
     """
     id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    product_code = models.CharField(max_length=20)
-    data = HStoreField(null=True, blank=True)
+    amount = models.DecimalField(max_digits=5, decimal_places=2)
```
Comment (gsvr) [inline recharges/models.py]: This will allow a max amount of R999.99.  Set max_digits to 9 as discussed with Mike.

**Comment by Lindani** [comment]
@gsvr Please review

File: recharges/models.py [hunk 10]
```diff
@@ -1,19 +1,18 @@
...
-    data = HStoreField(null=True, blank=True)
+    amount = models.DecimalField(max_digits=9, decimal_places=2)
+    msisdn = models.CharField(max_length=20)
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
 
     def __str__(self):  # __unicode__ on Python 2
-        return str(self.id)
\ No newline at end of file
+        return "R%s recharge for %s" % (self.amount, self.msisdn)
```
Comment (imsickofmaps) [inline recharges/models.py]: Drop the `R` please as that assumes currency of Rand which isn't defined in the system yet

File: recharges/tests.py [hunk 11]
```diff
@@ -43,10 +43,10 @@ def test_login(self):
...
 
-    def test_create_dummy_model_data(self):
+    def test_create_recharge_model_data(self):
         post_data = {
-            "product_code": "test_code",
-            "data": {'a': 'a', 'b': 2}
+            "amount": "10.0",
+            "msisdn": "084 123 4023"
         }
         response = self.client.post('/recharges/dummy/',
```
Comment (imsickofmaps) [inline recharges/tests.py]: `dummy` ref

File: recharges/tests.py [hunk 12]
```diff
@@ -54,6 +54,20 @@ def test_create_dummy_model_data(self):
...
+        d = Recharge.objects.last()
+        self.assertEqual(d.amount, 10.0)
+        self.assertEqual(d.msisdn, "084 123 4023")
+
+    def test_create_recharge_bad_model_data(self):
+        post_data = {
+            "amount": "109888787766650.00",
+            "msisdn": "084 123 4023"
+        }
+        response = self.client.post('/recharges/dummy/',
```
Comment (imsickofmaps) [inline recharges/tests.py]: `dummy` ref

File: recharges/urls.py [hunk 13]
```diff
@@ -5,7 +5,7 @@
 router = routers.DefaultRouter()
 router.register(r'users', views.UserViewSet)
 router.register(r'groups', views.GroupViewSet)
-router.register(r'dummy', views.DummyModelViewSet)
+router.register(r'dummy', views.RechargeViewSet)
```
Comment (imsickofmaps) [inline recharges/urls.py]: `dummy` ref

**Comment by imsickofmaps** [comment]
Getting there! @gsvr can you guide setting the api structure to `/api/v1/recharge`?

**Comment by imsickofmaps** [comment]
Also, needs a `.travis.yaml` so I can hook it up to Travis

**Comment by imsickofmaps** [comment]
:+1:

**Pull request merged**